contribute/style-guides/e2e-playwright.md

```diff
@@ -0,0 +1,299 @@
+
+    await page.getByTestId(selectors.pages.Login.username).fill(grafanaAPICredentials.user);
+    await page.getByTestId(selectors.pages.Login.password).fill(grafanaAPICredentials.password);
+
+    await page.getByTestId(selectors.pages.Login.submit).click();
+
+    await expect(page.getByTestId(selectors.components.NavToolbar.commandPaletteTrigger)).toBeVisible();
+  }
+);
+```
+
+### Advanced example
+
+Let's take a look at an example that uses the same selector for multiple items in a list for instance. In this example app, there's a list of data sources that we want to click on during an E2E test.
+
+```jsx
+<ul>
+  {dataSources.map(({ id, name }) => (
+    <li className="card-item-wrapper" key={id}>
+      <a className="card-item" href={`datasources/edit/${id}`}>
+        <div className="card-item-name">{name}</div>
+      </a>
+    </li>
+  ))}
+</ul>
+```
+
+Like in the basic example, start by creating a page abstraction:
+
+```typescript
+export const versionedPages = {
+[...]
+  DataSources: {
+    url: {
+      [MIN_GRAFANA_VERSION]: '/datasources',
+    },
+    dataSources: {
+      [MIN_GRAFANA_VERSION]: (dataSourceName: string) => `Data source list item ${dataSourceName}`,
+    },
+  },
+[...]
+};
+```
+
+You might have noticed that instead of a simple string as the selector, there's a function that takes a string parameter as an argument and returns a formatted string using the argument.
+
+Just as before, you need to add the `DataSources` url to the exported const `versionedPages` in `packages/grafana-e2e-selectors/src/selectors/pages.ts`.
+
+The next step is to use the `dataSources` selector function as in the following example:
+
+```jsx
+<ul>
+  {dataSources.map(({ id, name }) => (
+    <li className="card-item-wrapper" key={id}>
+      <a className="card-item" href={`datasources/edit/${id}`}>
+        <div className="card-item-name" data-testid={selectors.pages.DataSources.dataSources(name)}>
+          {name}
+        </div>
+      </a>
+    </li>
+  ))}
+</ul>
+```
+
+When this list is rendered with the data sources with names `A`, `B` and `C` ,the resulting HTML looks like this:
+
+```html
+<div class="card-item-name" data-testid="data-testid Data source list item A">A</div>
+<div class="card-item-name" data-testid="data-testid Data source list item B">B</div>
+<div class="card-item-name" data-testid="data-testid Data source list item C">C</div>
+```
+
+Now we can write our test. The one thing that differs from the previous [basic example](#basic-example) is that we pass in which data source we want to click as an argument to the selector function:
+
+```typescript
+test(
+  'List test clicks on data source named B',
+  {
+    tag: ['@datasources'],
+  },
+  async ({ selectors, page }) => {
+    await page.goto(selectors.pages.Datasources.url);
+    // Select datasource
+    const dataSource = 'B';
+    await page.getByTextId(dataSource).click();
+  }
+);
+```
+
+### aria-label versus data-testid
```

Comment:
i think we should remove this entire section. we no longer recommend anyone using `aria-label` for an e2e selector